From 22203141e5b10e6f2a9318f948b94b7aed947478 Mon Sep 17 00:00:00 2001
From: fairbird <rrrr53@hotmail.com>
Date: Sat, 10 May 2025 18:15:46 +0300
Subject: [PATCH] patch

---
 plugin.py | 54 +++++++++++-------------------------------------------
 1 file changed, 11 insertions(+), 43 deletions(-)

diff --git a/EPGImport/plugin.py b/EPGImport/plugin.py
index 1e53125..d4ac9e0 100644
--- a/EPGImport/plugin.py
+++ b/EPGImport/plugin.py
@@ -14,8 +14,6 @@ from . import EPGImport
 from . import EPGConfig
 
 
-from Components.SystemInfo import BoxInfo
-
 # Config
 from Components.ActionMap import ActionMap
 from Components.Button import Button
@@ -64,7 +62,6 @@ BouquetChannelListList = None
 serviceIgnoreList = None
 filterCounter = 0
 isFilterRunning = 0
-IMAGEDISTRO = BoxInfo.getItem("distro")
 
 SOURCE_LINKS = {
 	0: "https://github.com/oe-alliance/EPGimport-Sources/archive/refs/heads/main.tar.gz",
@@ -87,10 +84,9 @@ config.plugins.epgimport.repeat_import = ConfigInteger(default=0, limits=(0, 23)
 config.plugins.epgimport.runboot_restart = ConfigYesNo(default=False)
 config.plugins.epgimport.runboot_day = ConfigYesNo(default=False)
 config.plugins.epgimport.wakeup = ConfigClock(default=calcDefaultStarttime())
-# Different default in OpenATV:
-config.plugins.epgimport.showinplugins = ConfigYesNo(default=IMAGEDISTRO != "openatv")
+config.plugins.epgimport.showinplugins = ConfigYesNo(default=True)
 config.plugins.epgimport.showinextensions = ConfigYesNo(default=True)
-# config.plugins.epgimport.showinmainmenu = ConfigYesNo(default=False)
+config.plugins.epgimport.showinmainmenu = ConfigYesNo(default=False)
 config.plugins.epgimport.deepstandby = ConfigSelection(
 	default="skip",
 	choices=[
@@ -730,6 +726,13 @@ def main(session, **kwargs):
 	session.openWithCallback(doneConfiguring, EPGImportConfig)
 
 
+def main_menu(menuid, **kwargs):
+	if menuid == "mainmenu" and config.plugins.epgimport.showinmainmenu.getValue():
+		return [(_("EPG Importer"), start_import, "epgimporter", 45)]
+	else:
+		return []
+
+
 def doneConfiguring(*retVal):
 	"""user has closed configuration, check new values...."""
 	if autoStartTimer is not None:
@@ -1133,29 +1136,6 @@ extDescriptor = PluginDescriptor(name=_("EPG-Importer Now"), description=descrip
 pluginlist = PluginDescriptor(name=_("EPG-Importer"), description=description, where=PluginDescriptor.WHERE_PLUGINMENU, icon="plugin.png", fnc=main)
 
 
-def epgmenu(menuid, **kwargs):
-	if IMAGEDISTRO in ("openvix", "openbh", "ventonsupport", "egami", "openhdf", "opendroid"):
-		if menuid == "epg":
-			return [(_("EPG-Importer"), main, "epgimporter", 1002)]
-		else:
-			return []
-	elif IMAGEDISTRO in ("openatv"):
-		if menuid == "epg":
-			return [(_("EPG-Importer"), main, "epgimporter", None)]
-		else:
-			return []
-	elif IMAGEDISTRO in ("teamblue"):
-		if menuid == "epg_menu":
-			return [(_("EPG-Importer"), main, "epgimporter", 95)]
-		else:
-			return []
-	else:
-		if menuid == "setup":
-			return [(_("EPG-Importer"), main, "epgimporter", 1002)]
-		else:
-			return []
-
-
 def Plugins(**kwargs):
 	result = [
 		PluginDescriptor(
@@ -1172,7 +1152,7 @@ def Plugins(**kwargs):
 			name=_("EPG-Importer"),
 			description=description,
 			where=PluginDescriptor.WHERE_MENU,
-			fnc=epgmenu
+			fnc=main_menu
 		),
 	]
 	if config.plugins.epgimport.showinextensions.value:
