From 94a2149807a6450a14867daa604678058b21a489 Mon Sep 17 00:00:00 2001
From: fairbird <rrrr53@hotmail.com>
Date: Wed, 28 Feb 2024 13:31:01 +0300
Subject: [PATCH] patch

---
 plugin.py | 54 +++++++++++-------------------------------------------
 1 file changed, 11 insertions(+), 43 deletions(-)

diff --git a/EPGImport/plugin.py b/EPGImport/plugin.py
index 22cef53..935a1aa 100644
--- a/EPGImport/plugin.py
+++ b/EPGImport/plugin.py
@@ -13,13 +13,6 @@ import time
 
 import enigma
 
-try:
-    from Components.SystemInfo import BoxInfo
-    IMAGEDISTRO = BoxInfo.getItem("distro")
-except:
-    from boxbranding import getImageDistro
-    IMAGEDISTRO = getImageDistro()
-
 # Config
 from Components.ActionMap import ActionMap
 from Components.Button import Button
@@ -63,7 +56,7 @@ def calcDefaultStarttime():
 
 #Set default configuration
 config.plugins.epgimport = ConfigSubsection()
-config.plugins.epgimport.enabled = ConfigEnableDisable(default=False)
+config.plugins.epgimport.enabled = ConfigEnableDisable(default=True)
 config.plugins.epgimport.runboot = ConfigSelection(default="4", choices=[
 		("1", _("always")),
 		("2", _("only manual boot")),
@@ -72,10 +65,8 @@ config.plugins.epgimport.runboot = ConfigSelection(default="4", choices=[
 		])
 config.plugins.epgimport.runboot_restart = ConfigYesNo(default=False)
 config.plugins.epgimport.runboot_day = ConfigYesNo(default=False)
-config.plugins.epgimport.wakeupsleep = ConfigEnableDisable(default=False)
 config.plugins.epgimport.wakeup = ConfigClock(default=calcDefaultStarttime())
-# Different default in OpenATV:
-config.plugins.epgimport.showinplugins = ConfigYesNo(default=IMAGEDISTRO != "openatv")
+config.plugins.epgimport.showinplugins = ConfigYesNo(default=True)
 config.plugins.epgimport.showinextensions = ConfigYesNo(default=True)
 config.plugins.epgimport.deepstandby = ConfigSelection(default="skip", choices=[
 		("wakeup", _("wake up and import")),
@@ -84,7 +75,7 @@ config.plugins.epgimport.deepstandby = ConfigSelection(default="skip", choices=[
 config.plugins.epgimport.standby_afterwakeup = ConfigYesNo(default=False)
 config.plugins.epgimport.shutdown = ConfigYesNo(default=False)
 config.plugins.epgimport.longDescDays = ConfigNumber(default=5)
-#config.plugins.epgimport.showinmainmenu = ConfigYesNo(default = False)
+config.plugins.epgimport.showinmainmenu = ConfigYesNo(default = False)
 config.plugins.epgimport.deepstandby_afterimport = NoSave(ConfigYesNo(default=False))
 config.plugins.epgimport.parse_autotimer = ConfigYesNo(default=False)
 config.plugins.epgimport.import_onlybouquet = ConfigYesNo(default=False)
@@ -377,7 +368,7 @@ class EPGImportConfig(ConfigListScreen, Screen):
 		self.cfg_runboot_restart = getConfigListEntry(_("Skip import on restart GUI"), self.EPG.runboot_restart)
 		self.cfg_showinextensions = getConfigListEntry(_("Show \"EPGImport\" in extensions"), self.EPG.showinextensions)
 		self.cfg_showinplugins = getConfigListEntry(_("Show \"EPGImport\" in plugins"), self.EPG.showinplugins)
-#		self.cfg_showinmainmenu = getConfigListEntry(_("Show \"EPG Importer\" in main menu"), self.EPG.showinmainmenu)
+		self.cfg_showinmainmenu = getConfigListEntry(_("Show \"EPG Importer\" in main menu"), self.EPG.showinmainmenu)
 		self.cfg_longDescDays = getConfigListEntry(_("Load long descriptions up to X days"), self.EPG.longDescDays)
 		self.cfg_parse_autotimer = getConfigListEntry(_("Run AutoTimer after import"), self.EPG.parse_autotimer)
 		self.cfg_clear_oldepg = getConfigListEntry(_("Clearing current EPG before import"), config.plugins.epgimport.clear_oldepg)
@@ -399,6 +390,7 @@ class EPGImportConfig(ConfigListScreen, Screen):
 				list.append(self.cfg_runboot_restart)
 		list.append(self.cfg_showinextensions)
 		list.append(self.cfg_showinplugins)
+		list.append(self.cfg_showinmainmenu)
 		list.append(self.cfg_import_onlybouquet)
 		list.append(self.cfg_import_onlyiptv)
 		if hasattr(enigma.eEPGCache, 'flushEPG'):
@@ -753,11 +745,11 @@ def start_import(session, **kwargs):
 def main(session, **kwargs):
 	session.openWithCallback(doneConfiguring, EPGImportConfig)
 
-#def main_menu(menuid, **kwargs):
-#	if menuid == "mainmenu" and config.plugins.epgimport.showinmainmenu.getValue():
-#		return [(_("EPG Importer"), start_import, "epgimporter", 45)]
-#	else:
-#		return []
+def main_menu(menuid, **kwargs):
+	if menuid == "mainmenu" and config.plugins.epgimport.showinmainmenu.getValue():
+		return [(_("EPG Importer"), start_import, "epgimporter", 45)]
+	else:
+		return []
 
 
 def doneConfiguring(session, retval):
@@ -1095,30 +1087,6 @@ config.plugins.epgimport.showinextensions.addNotifier(setExtensionsmenu, initial
 extDescriptor = PluginDescriptor(name=_("EPG-Importer"), description=description, where=PluginDescriptor.WHERE_EXTENSIONSMENU, fnc=extensionsmenu)
 pluginlist = PluginDescriptor(name=_("EPG-Importer"), description=description, where=PluginDescriptor.WHERE_PLUGINMENU, icon='plugin.png', fnc=main)
 
-
-def epgmenu(menuid, **kwargs):
-	if IMAGEDISTRO in ("openvix", "openbh", "ventonsupport", "egami", "openhdf", "opendroid"):
-		if menuid == "epg":
-			return [(_("EPG-Importer"), main, "epgimporter", 1002)]
-		else:
-			return []
-	elif IMAGEDISTRO in ("openatv"):
-		if menuid == "epg":
-			return [(_("EPG-Importer"), main, "epgimporter", None)]
-		else:
-			return []
-	elif IMAGEDISTRO in ("teamblue"):
-		if menuid == "epg_menu":
-			return [(_("EPG-Importer"), main, "epgimporter", 95)]
-		else:
-			return []
-	else:
-		if menuid == "setup":
-			return [(_("EPG-Importer"), main, "epgimporter", 1002)]
-		else:
-			return []
-
-
 def Plugins(**kwargs):
 	result = [
 		PluginDescriptor(
@@ -1135,7 +1103,7 @@ def Plugins(**kwargs):
 			name=_("EPG-Importer"),
 			description=description,
 			where=PluginDescriptor.WHERE_MENU,
-			fnc=epgmenu
+			fnc=main_menu
 		),
 	]
 	if config.plugins.epgimport.showinextensions.value:
-- 
2.40.1

