From 3541a37c1e9f036453c9dbe3953870427a906af5 Mon Sep 17 00:00:00 2001
From: fairbird <rrrr53@hotmail.com>
Date: Mon, 12 Jul 2021 01:46:03 +0300
Subject: Add plugin in networksetup

---
 .../Plugins/Extensions/InternetSpeedTest/plugin.py  | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py b/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py
index 937b457..86c835d 100644
--- a/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py
+++ b/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py
@@ -133,15 +133,12 @@ class internetspeedtest(Screen):
         self.close()
 
 
-def main(session, **kwargs):
+def main(session, iface):
     session.open(internetspeedtest)
 
+def callFunction(iface):
+    return main
+
 
 def Plugins(**kwargs):
-    list = []
-    if HD.width() > 1280:
-        list.append(PluginDescriptor(name=("Internet Speed Test"), description=_("Test your internet speed"), where=PluginDescriptor.WHERE_PLUGINMENU, icon="speedtestfhd.png", fnc=main))
-    else:
-        list.append(PluginDescriptor(name=("Internet Speed Test"), description=_("Test your internet speed"), where=PluginDescriptor.WHERE_PLUGINMENU, icon="speedtest.png", fnc=main))
-    list.append(PluginDescriptor(name=("Internet Speed Test"), description=_("Test your internet speed"), where=PluginDescriptor.WHERE_EXTENSIONSMENU, fnc=main))
-    return list
+    return PluginDescriptor(name=_("InternetSpeedTest"), description=_("Internet Speed Test") + "\n", where=PluginDescriptor.WHERE_NETWORKSETUP, needsRestart=False, fnc={"ifaceSupported": callFunction, "menuEntryName": lambda x: _("InternetSpeedTest"), "menuEntryDescription": lambda x: _("Internet Speed Test...") + "\n"})
