From 3d7f5a4bdb5a138bdb21d1e329bf259b9e974f27 Mon Sep 17 00:00:00 2001
From: fairbird <rrrr53@hotmail.com>
Date: Wed, 14 Jul 2021 23:54:27 +0300
Subject: Get rid of OV Team stuff

---

diff --git a/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py b/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py
index 937b457..86c835d 100644
--- a/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py
+++ b/usr/lib/enigma2/python/Plugins/Extensions/InternetSpeedTest/plugin.py
@@ -6,7 +6,7 @@ class internetspeedtest(Screen):
 from Components.Button import Button
 from os import remove, environ, chmod, path
 import gettext
-from enigma import getDesktop, eConsoleAppContainer, getPyExt
+from enigma import getDesktop, eConsoleAppContainer
 from Components.Language import language
 from Tools.Directories import resolveFilename, SCOPE_LANGUAGE, SCOPE_PLUGINS
 from Plugins.Plugin import PluginDescriptor
@@ -14,10 +14,10 @@ class internetspeedtest(Screen):
 from Components.ActionMap import ActionMap
 from Components.Label import Label
 
-plugin_path = resolveFilename(SCOPE_PLUGINS, "Extensions/InternetSpeedTest/speedtest.")
+plugin_path = resolveFilename(SCOPE_PLUGINS, "Extensions/InternetSpeedTest/speedtest.py")
 font = resolveFilename(SCOPE_PLUGINS, "Extensions/InternetSpeedTest/fonts")
 skin_path = resolveFilename(SCOPE_PLUGINS, "Extensions/InternetSpeedTest/")
-cmd = "python " + plugin_path + getPyExt()
+cmd = 'python ' + plugin_path + ' --no-pre-allocate --share'
 if path.exists(plugin_path):
     chmod(plugin_path, 0o755)
 HD = getDesktop(0).size()
