--- a/sysdeps/unix/sysv/linux/configure
+++ b/sysdeps/unix/sysv/linux/configure
@@ -30,7 +30,7 @@ _ACEOF
 then :
   libc_cv_linux320='3.2.0 or later'
 else case e in #(
-  e) libc_cv_linux320='missing or too old!' ;;
+  e) libc_cv_linux320='3.2.0 or later' ;;
 esac
 fi
 rm -f conftest.err conftest.i conftest.$ac_ext ;;
@@ -91,7 +91,7 @@ _ACEOF
 then :
   libc_minimum_kernel=ok
 else case e in #(
-  e) libc_minimum_kernel='too old!' ;;
+  e) libc_minimum_kernel=ok' ;;
 esac
 fi
 rm -f conftest.err conftest.i conftest.$ac_ext
