From 60c650333ef8df54e0c2018e5c42bc3330dba543 Mon Sep 17 00:00:00 2001
From: fairbird <rrrr53@hotmail.com>
Date: Tue, 21 May 2024 16:49:58 +0300
Subject: [PATCH] no-need-to-check-depends

---

diff --git a/IPTVPlayer/components/iptvconfigmenu.py b/IPTVPlayer/components/iptvconfigmenu.py
index b42045d..199cb68 100755
--- a/IPTVPlayer/components/iptvconfigmenu.py
+++ b/IPTVPlayer/components/iptvconfigmenu.py
@@ -224,7 +224,7 @@ config.plugins.iptvplayer.myjd_password = ConfigText(default="", fixed_size=Fals
 config.plugins.iptvplayer.api_key_youtube = ConfigText(default="", fixed_size=False)
 
 # Update
-config.plugins.iptvplayer.autoCheckForUpdate = ConfigYesNo(default=True)
+config.plugins.iptvplayer.autoCheckForUpdate = ConfigYesNo(default=False)
 config.plugins.iptvplayer.updateLastCheckedVersion = ConfigText(default="00.00.00.00", fixed_size=False)
 config.plugins.iptvplayer.fakeUpdate = ConfigSelection(default="fake", choices=[("fake", "  ")])
 config.plugins.iptvplayer.downgradePossible = ConfigYesNo(default=False)
diff --git a/IPTVPlayer/components/iptvextmovieplayer.py b/IPTVPlayer/components/iptvextmovieplayer.py
index 53c673a..d9a28c3 100755
--- a/IPTVPlayer/components/iptvextmovieplayer.py
+++ b/IPTVPlayer/components/iptvextmovieplayer.py
@@ -2023,8 +2023,7 @@ class IPTVExtMoviePlayer(Screen):
                             cmd += (' "proxy=%s" ' % tmp)
             cmd += " > /dev/null"
         else:
-            exteplayer3path = config.plugins.iptvplayer.exteplayer3path.value
-            cmd = exteplayer3path
+            cmd = '/usr/bin/exteplayer3'
             tmpUri = strwithmeta(self.fileSRC)
 
             audioUri = ''
diff --git a/IPTVPlayer/tools/e2ijs.py b/IPTVPlayer/tools/e2ijs.py
index 3bbe609..6384b50 100755
--- a/IPTVPlayer/tools/e2ijs.py
+++ b/IPTVPlayer/tools/e2ijs.py
@@ -39,7 +39,7 @@ def duktape_execute(cmd_params):
         noDuk = True
         ret['code'] = 127
 
-    if noDuk:
+    if noDuk and not fileExists("/usr/lib/libduktape.so.207"):
         messages = [_('The %s utility is necessary here but it was not detected.') % ('duktape')]
         messages.append(_('Please consider restart your Engima2 and agree to install the %s utlity when the %s will propose this.') % ('duktape', 'E2iPlayer'))
         GetIPTVNotify().push('\n'.join(messages), 'error', 40, 'no_duktape', 40)
-- 
2.43.0

