From: fairbird, RAED <rrrr53@hotmail.com>
Date: Fri, 10 Mar 2021 14:55:06 +0300
Subject: Fix PLAYPAUSE Key to start Timshift

-----------------------------------------

diff -Naur a/configure.ac   b/configure.ac 
--- a/configure.ac  	2019-10-21 03:46:30.425352000 +0330
+++ b/configure.ac 	2020-01-31 16:27:24.846660642 +0330
@@ -496,7 +496,7 @@ if test "$BOXBRAND" == "dags"; then
 	AC_DEFINE(FORCE_ADVANCED_REMOTE, 1,[define to fixup the input device identification when the remote control is actually an 'advanced' remote (with play/forward/rewind keys)])
 fi
 
-if test "$BOXBRAND" == "dreambox" && "$STBPLATFORM" != "dmamlogic"; then
+if test "$BOXBRAND" == "dreambox"; then
 	AC_DEFINE(DREAMBOX, 1, [Dreambox hardware])
 	AC_DEFINE(KEY_PLAY_ACTUALLY_IS_KEY_PLAYPAUSE, 1,[define when rc sends a KEY_PLAY event for its KEY_PLAYPAUSE key])
 fi
