diff --git a/lib/python/Screens/Information.py b/lib/python/Screens/Information.py
index 0b76d0715..c19b567fd 100644
--- a/lib/python/Screens/Information.py
+++ b/lib/python/Screens/Information.py
@@ -446,12 +446,6 @@ class ImageInformation(InformationBase):
 		info = []
 		info.append(formatLine("M", _("Copyright %s 2018-%s Team OpenVision") % (u"\u00A9", localtime()[0])))
 		info.append("")
-		info.append(formatLine("M", _("OpenVision is an open source project with no commercial funding.")))
-		info.append(formatLine("M", _("The team relies on donations to fund OpenVision development.")))
-		info.append(formatLine("M", _("If you would like to support us then please consider donating.")))
-		info.append("")
-		info.append(formatLine("M", _("Donate at %s") % "https://forum.openvision.tech/app.php/donate"))
-		info.append("")
 		info.append(formatLine("P1", _("OpenVision version"), BoxInfo.getItem("imgversion")))
 		info.append(formatLine("P1", _("OpenVision revision"), BoxInfo.getItem("imgrevision")))
 		if config.misc.OVupdatecheck.value:
