From: fairbird, RAED <rrrr53@hotmail.com>
Date: Thu, 11 Mar 2021 14:55:06 +0300
Subject: Revert the commmit that deleted HardwareInfo.py file

---

diff -Naur a/lib/python/Tools/Makefile.am  b/lib/python/Tools/Makefile.am 
--- a/lib/python/Tools/Makefile.am 	2018-02-16 10:23:52.426216568 +0100
+++ b/lib/python/Tools/Makefile.am 	2018-02-16 10:25:30.325492539 +0100
@@ -4,7 +4,7 @@ install_PYTHON = \
 	FuzzyDate.py XMLTools.py Directories.py NumericalTextInput.py \
 	KeyBindings.py BoundFunction.py ISO639.py Notifications.py __init__.py \
 	RedirectOutput.py StbHardware.py Import.py Event.py CList.py CIHelper.py \
-	LoadPixmap.py Profile.py Transponder.py ASCIItranslit.py \
+	LoadPixmap.py Profile.py HardwareInfo.py Transponder.py ASCIItranslit.py \
 	Downloader.py Trashcan.py GetEcmInfo.py Alternatives.py TextBoundary.py \
 	camcontrol.py CountryCodes.py Multiboot.py FallbackTimer.py \
 	CopyFiles.py Log.py LogConfig.py Geolocation.py UnitConversions.py Trace.py WeatherID.py
diff --git a/lib/python/Tools/HardwareInfo.py b/lib/python/Tools/HardwareInfo.py
new file mode 100644
index 0000000..b07fe77
--- /dev/null
+++ b/lib/python/Tools/HardwareInfo.py
@@ -0,0 +1,66 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from boxbranding import getHaveHDMI
+
+hw_info = None
+
+
+class HardwareInfo:
+	device_name = _("unavailable")
+	device_brandname = None
+	device_model = None
+	device_version = ""
+	device_revision = ""
+	device_hdmi = False
+
+	def __init__(self):
+		global hw_info
+		if hw_info:
+			return
+		hw_info = self
+
+		print("[HardwareInfo] Scanning hardware info")
+		# Version
+		try:
+			print("[HardwareInfo] Read /proc/stb/info/version")
+			self.device_version = open("/proc/stb/info/version").read().strip()
+		except:
+			print("[HardwareInfo] Read /proc/stb/info/version failed.")
+
+		self.device_model = self.device_model or self.device_name
+		self.device_hw = self.device_model
+		self.machine_name = self.device_model
+
+		if self.device_revision:
+			self.device_string = "%s (%s-%s)" % (self.device_hw, self.device_revision, self.device_version)
+		elif self.device_version:
+			self.device_string = "%s (%s)" % (self.device_hw, self.device_version)
+		else:
+			self.device_string = self.device_hw
+
+		# only some early DMM boxes do not have HDMI hardware
+		self.device_hdmi = getHaveHDMI() == "True"
+
+		print("[HardwareInfo] Detected: " + self.get_device_string())
+
+	def get_device_name(self):
+		return hw_info.device_name
+
+	def get_device_model(self):
+		return hw_info.device_model
+
+	def get_device_version(self):
+		return hw_info.device_version
+
+	def get_device_revision(self):
+		return hw_info.device_revision
+
+	def get_device_string(self):
+		return hw_info.device_string
+
+	def get_machine_name(self):
+		return hw_info.machine_name
+
+	def has_hdmi(self):
+		return hw_info.device_hdmi
