From: fairbird, RAED <rrrr53@hotmail.com>
Date: Thu, 11 Mar 2021 14:55:06 +0300
Subject: Revert the commmit that deleted HardwareInfo.py file

---

diff -Naur a/lib/python/Tools/Makefile.am  b/lib/python/Tools/Makefile.am 
--- a/lib/python/Tools/Makefile.am 	2018-02-16 10:23:52.426216568 +0100
+++ b/lib/python/Tools/Makefile.am 	2018-02-16 10:25:30.325492539 +0100
@@ -4,7 +4,7 @@ install_PYTHON = \
 	FuzzyDate.py XMLTools.py Directories.py NumericalTextInput.py \
 	KeyBindings.py BoundFunction.py ISO639.py Notifications.py __init__.py \
 	RedirectOutput.py StbHardware.py Import.py Event.py CList.py CIHelper.py \
-	LoadPixmap.py Profile.py Transponder.py ASCIItranslit.py \
+	LoadPixmap.py Profile.py HardwareInfo.py Transponder.py ASCIItranslit.py \
 	Downloader.py Trashcan.py GetEcmInfo.py Alternatives.py TextBoundary.py \
 	camcontrol.py CountryCodes.py Multiboot.py FallbackTimer.py \
 	CopyFiles.py Log.py LogConfig.py Geolocation.py UnitConversions.py Trace.py WeatherID.py
