diff --git a/data/menu.xml b/data/menu.xml
index afd718c96..45006d9d9 100644
--- a/data/menu.xml
+++ b/data/menu.xml
@@ -102,5 +102,6 @@ self.session.openWithCallback(msgClosed, FactoryReset)
 			<item text="Restart enigma in debug mode" entryID="restart_enigma_debug"><screen module="Standby" screen="TryQuitMainloop">6</screen></item>
 			<item text="Deep standby" requires="DeepstandbySupport" entryID="deep_standby"><screen module="Standby" screen="TryQuitMainloop">1</screen></item>
 			<item text="Shutdown" requires="!DeepstandbySupport" entryID="deep_standby"><screen module="Standby" screen="TryQuitMainloop">1</screen></item>
+			<item text="Recovery Mode" requires="RecoveryMode" entryID="maintenance_mode"><screen module="Standby" screen="TryQuitMainloop">16</screen></item>
 		</menu>
 </menu>
diff --git a/lib/python/Components/Lcd.py b/lib/python/Components/Lcd.py
index c40efa2b0..55ba60f99 100644
--- a/lib/python/Components/Lcd.py
+++ b/lib/python/Components/Lcd.py
@@ -3,6 +3,7 @@ from enigma import eDBoxLCD
 from Components.SystemInfo import SystemInfo
 from Screens.InfoBar import InfoBar
 from Screens.Screen import Screen
+from enigma import getBoxType
 
 class dummyScreen(Screen):
 	skin = """<screen position="0,0" size="0,0" transparent="1">
@@ -75,7 +76,10 @@ def InitLcd():
 			config.lcd.contrast.addNotifier(setLCDcontrast);
 		else:
 			config.lcd.contrast = ConfigNothing()
-			standby_default = 1
+			if getBoxType() in ('dm900','dm920'):
+				standby_default = 4
+			else:
+				standby_default = 1
 
 		config.lcd.standby = ConfigSlider(default=standby_default, limits=(0, 10))
 		config.lcd.standby.addNotifier(setLCDbright);
diff --git a/lib/python/Components/SystemInfo.py b/lib/python/Components/SystemInfo.py
index 042ad54ef..d6523abc2 100644
--- a/lib/python/Components/SystemInfo.py
+++ b/lib/python/Components/SystemInfo.py
@@ -98,3 +98,4 @@ SystemInfo["CanDownmixAC3"] = fileHas("/proc/stb/audio/ac3_choices", "downmix")
 SystemInfo["CanDownmixDTS"] = fileHas("/proc/stb/audio/dts_choices", "downmix")
 SystemInfo["CanDownmixAAC"] = fileHas("/proc/stb/audio/aac_choices", "downmix")
 SystemInfo["HDMIAudioSource"] = fileCheck("/proc/stb/hdmi/audio_source")
+SystemInfo["RecoveryMode"] = fileCheck("/proc/stb/fp/boot_mode")
diff --git a/lib/python/Screens/Standby.py b/lib/python/Screens/Standby.py
index 1740fd3bf..aac921620 100644
--- a/lib/python/Screens/Standby.py
+++ b/lib/python/Screens/Standby.py
@@ -207,6 +207,7 @@ class QuitMainloopScreen(Screen):
 			4: _("Your frontprocessor will be upgraded\nPlease wait until your receiver reboots\nThis may take a few minutes"),
 			5: _("The user interface of your receiver is restarting\ndue to an error in mytest.py"),
 			6: _("The user interface of your receiver is restarting in debug mode"),
+			16: _("Your receiver is rebooting into Recovery Mode"),
 			42: _("Unattended upgrade in progress\nPlease wait until your receiver reboots\nThis may take a few minutes") }.get(retvalue)
 		self["text"] = Label(text)
 
@@ -244,6 +245,7 @@ class TryQuitMainloop(MessageBox):
 				3: _("Really restart now?"),
 				4: _("Really upgrade the frontprocessor and reboot now?"),
 				6: _("Really restart in debug mode now?"),
+				16: _("Really reboot into Recovery Mode?"),
 				42: _("Really upgrade your settop box and reboot now?") }.get(retvalue)
 			if text:
 				MessageBox.__init__(self, session, reason+text, type = MessageBox.TYPE_YESNO, timeout = timeout, default = default_yes)
diff --git a/tools/enigma2.sh.in b/tools/enigma2.sh.in
index 6811a3672..1cdddb4be 100755
--- a/tools/enigma2.sh.in
+++ b/tools/enigma2.sh.in
@@ -85,6 +85,10 @@ while : ; do
 		6)
 			DEBUG_LVL=4
 			;;
+		16)
+			echo "rescue" > /proc/stb/fp/boot_mode
+			/sbin/reboot
+			;;
 		42)
 			df -P | grep -v "tmpfs " | awk '{print $6}' | tail -n +3 > /tmp/upgrade_mountpoints.txt
 			while read line; do
