From: fairbird <rrrr53@gmail.com>
Date: Tue, 23 May 2022 13:54:00 +0200
Subject: [PATCH] configure: fix check for distutils

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 75ae328..6447e8d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -16,8 +16,6 @@ AC_ARG_WITH(gstversion,
 
 PKG_CHECK_MODULES(GSTREAMER, gstreamer-$GST_MAJORMINOR gstreamer-pbutils-$GST_MAJORMINOR)
 
-AX_CXX_COMPILE_STDCXX_17([ext],[mandatory])
-
 AC_DEFINE([DEBUG])
 
 AC_CONFIG_FILES([
